src/components/rootUserSpecific/NewBookingDialog.jsx

```diff
@@ -222,13 +224,26 @@ export function NewBookingDialog({ fetchUserBookings, username, email }) {
     setToggle("from");
   };
 
+  const findOverlappingRides = () => {
+    router.push({
```

Comment:
since you are using query strings to pass props, on clearing the filters in AllUserBookings, the query string is not cleared. Hence, on reloading the page or just switching tabs fills the filters again. (i.e. filters stay there forever, unless someone removes them from the query string)